### ISSUE TITLE
Proxy feature

### ISSUE THREAD

**Issue by TitouanS31** [nautobot/pynautobot#201]
It would be great to have a proxy option for api calls. We could hence do something like this.

```
proxies = {
    "http": "http://my_super_proxy.com:1234",
    "https": "https://my_super_proxy.com:1234",
}

nautobot = pynautobot.api(
    url = "https://demo.nautobot.com",
    token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    proxies = proxies
)
```

**Comment by glennmatthews** [comment]
As a workaround, you can probably set the standard `HTTP_PROXY` and `HTTPS_PROXY` environment variables.

**Comment by joewesch** [comment]
This should already be possible by [modifying the http session directly](https://docs.nautobot.com/projects/pynautobot/en/stable/dev/advanced/session/) after you instantiate the API class in the same way you modify headers or other items.

```python
nautobot = pynautobot.api(
    url = "https://demo.nautobot.com",
    token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
)
nautobot.http_session.proxies = {
    "http": "http://my_super_proxy.com:1234",
    "https": "https://my_super_proxy.com:1234",
}
```

**Comment by TitouanS31** [comment]
I though modifying the http session after the API instanciation would do the trick... but the api validation fails before exiting the call. I could catch the error and patch the session after all but I think that is not very clean. For now, I have just added 2 lines to the API `__init__` function to handle the proxy.

**Comment by joewesch** [comment]
Yes, I thought of that after giving you that update. We really need to change that auto-validation to happen during your first API call rather than on instantiation.

**Comment by joewesch** [comment]
This should now be possible as described in [my previous comment](https://github.com/nautobot/pynautobot/issues/201#issuecomment-2176072519).
